### PR TITLE
L1T phase-2: modify HPS taus

### PR DESCRIPTION
**Pull request by cecilecaillol**
#### PR description:

This is another PR to port all L1T Phase-2 developments to cms-sw. This PR updates the HPS and Puppi tau constructors. It includes several PRs made locally, the details of which can be found in the twiki https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideL1TPhase2Instructions and in the cms-l1t-offline branch.

### CODE REVIEW

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
+l1

File: DataFormats/L1TParticleFlow/interface/HPSPFTau.h [hunk 1]
```diff
@@ -67,6 +67,7 @@ namespace l1t {
     float sumNeutralIso() const { return sumNeutralIso_; }
     float sumCombinedIso() const { return sumCombinedIso_; }
     float sumChargedIsoPileup() const { return sumChargedIsoPileup_; }
+    float Z() const { return Z_; }
```
Comment (perrotta) [inline DataFormats/L1TParticleFlow/interface/HPSPFTau.h:70]: Class members and methods should have lower case initials
```suggestion
    float z() const { return z_; }
```

File: DataFormats/L1TParticleFlow/interface/HPSPFTau.h [hunk 2]
```diff
@@ -133,6 +134,7 @@ namespace l1t {
     void setSumNeutralIso(float sumNeutralIso) { sumNeutralIso_ = sumNeutralIso; }
     void setSumCombinedIso(float sumCombinedIso) { sumCombinedIso_ = sumCombinedIso; }
     void setSumChargedIsoPileup(float sumChargedIsoPileup) { sumChargedIsoPileup_ = sumChargedIsoPileup; }
+    void setZ(float Z) { Z_ = Z; }
```
Comment (perrotta) [inline DataFormats/L1TParticleFlow/interface/HPSPFTau.h:137]:
```suggestion
    void setZ(float Z) { z_ = Z; }
```

**Comment by cecilecaillol** [comment]
please test

File: DataFormats/L1TParticleFlow/src/classes_def.xml [hunk 3]
```diff
@@ -44,9 +44,11 @@
   <class name="edm::RefVector<l1t::PFTauCollection>" /> 
   <class name="std::vector<edm::Ref<l1t::PFTauCollection> >" /> 
 
-  <class name="l1t::HPSPFTau" ClassVersion="12">
-    <version ClassVersion="11" checksum="98678086"/>
+  <class name="l1t::HPSPFTau" ClassVersion="14">
+    <version ClassVersion="14" checksum="588131857"/>
+    <version ClassVersion="13" checksum="588108529"/>
```
Comment (perrotta) [inline DataFormats/L1TParticleFlow/src/classes_def.xml:49]: One class version update per PR is enough:
```suggestion
  <class name="l1t::HPSPFTau" ClassVersion="13">
    <version ClassVersion="13" checksum="588131857"/>
```

**Comment by AdrianoDee** [comment]
please test

**Comment by cecilecaillol** [comment]
+l1

**Comment by srimanob** [comment]
+Upgrade

**Pull request merged**